src/http/server.cr

```diff
@@ -95,72 +100,187 @@ class HTTP::Server
   {% end %}
 
   @wants_close = false
+  @interfaces = [] of Socket::Server
 
-  def self.new(port, &handler : Context ->)
-    new("127.0.0.1", port, &handler)
+  # DEPRECATED: Create an instance and call `#bind` instead.
```

Comment:
We don't deprecate before 1.0, we just remove. Unless it's super duper common and used in many places in one project (think `MemoryIO`), and even then we make the overloads print notices using macros otherwise nobody will notice.